src/problems/shinny_sam_sudoku.clj

```diff
@@ -0,0 +1,150 @@
+(ns problems.shinny-sam-sudoku)
+
+(defrecord State [board curr-position])
+
+;check row/column for duplicates
+(defn duplicates? [coll]
+  (not (apply distinct? (filter #(> % 0) coll))))
+
+;example to test all the other functions
+
+
+;check if a value/element is in a collection
+(defn in? [coll value]
+  (= true (some #(= value %) coll)))
+
+;return a row
+(defn row-seq [board y]
+  (nth board y))
+
+;return a column
+(defn col-seq [board x]
+  (for [y board]
+    (last(take (+ x 1) y))))
+
+;check if a row, column or 3x3 is legal
+(defn legal? [coll]
+  (and (not (duplicates? coll))))
+
+
+;the range of number(1-9) to put into the board
+(def numBank (range 1 10))
+
+;put numbers into the current position of the board
+(defn add [board position number numBank]
```

Comment:
I think you can probably get rid of `numBank` and just pass the right number in here rather than extracting it from `numBank`?